### ISSUE TITLE
Shape deletion does not delete record in database

### ISSUE THREAD

**Issue by timk1** [GeospatialPython/pyshp#57]
Deleting a shape causes shp and dbf to become mismatched
Recommend adding line:

```
elif shape and not part and not point:
            del self._shapes[shape]
            del self.records[shape]
```

**Comment by karimbahgat** [comment]
Would have worked, but no longer relevant after #98 which drops the Editor class.

**Comment by Weixing-Zhang** [comment]
Does anyone run into the same issue? 

Deleting a shape causes shp and dbf  mismatch problem even I am using the latest version of pyshp (1.2.2)
